### PR TITLE
#50 - Boyer-Moore string search algorithm added

### PR DESCRIPTION
**Pull request by bekesibeks**

### CODE REVIEW

File: javatests/com/williamfiset/algorithms/strings/BoyerMooreStringSearchTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    underTest = new BoyerMooreStringSearch();
+  }
+
+  @Test
+  public void shouldReturnEmptyListOnNullInput() {
+    assertThat(underTest.findOccurrences(null, "")).isEmpty();
+    assertThat(underTest.findOccurrences("", null)).isEmpty();
+    assertThat(underTest.findOccurrences(null, null)).isEmpty();
+  }
+
```
Comment (williamfiset) [inline javatests/com/williamfiset/algorithms/strings/BoyerMooreStringSearchTest.java]: Request: Can you add a few more tests? What is the expected behavior for an empty text? What about an empty pattern? A pattern longer than the text? No matches at all? etc..

File: com/williamfiset/algorithms/strings/BoyerMooreStringSearch.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.williamfiset.algorithms.strings;
+
+import static java.util.Objects.isNull;
+
```
Comment (williamfiset) [inline com/williamfiset/algorithms/strings/BoyerMooreStringSearch.java]: Can you update the README string section to add an entry for Boyer-Moore to point to this file?

**Comment by williamfiset** [comment]
Going ahead and merging this so that #121 can go through smooth, please submit another PR after #121 with the requested changes, that would be appreciated!
